domain/modeldefaults/service/service.go

```diff
@@ -29,11 +30,23 @@ type ModelDefaultsProvider interface {
 // ModelDefaultsProviderFunc is a func type that implements [ModelDefaultsProvider].
 type ModelDefaultsProviderFunc func(context.Context) (modeldefaults.Defaults, error)
 
+// ModelConfigProviderFunc describes a type that is able to return a
+// [environs.ModelConfigProvider] for the specified cloud type. If the no
+// model config provider exists for the supplied cloud type then a
+// [coreerrors.NotFound] error is returned. If the cloud type provider does not
+// support model config then a [coreerrors.NotSupported] error is returned.
+type ModelConfigProviderFunc func(string) (environs.ModelConfigProvider, error)
```

Comment:
question: Why use a string here rather than an actual type for a cloud? Same for the ModelCloudType return value.

Reply:
We don't have a dedicated type to represent this. We use strings everywhere. I agree it would be nice but I am avoiding the side quests at the moment.

Reply:
I didn't realize we had not types for clouds until later.